### PR TITLE
CNDIT-1753 Addition of record_status_cd constraint and Role for person_participation

### PR DESCRIPTION
**Pull request by upasanapattnaik-eq**
## Notes

Updates for person_participation and organization_participation json columns in organization stored procedure `sp_organization_event`. 
Additional columns for morbidity and lab test post-processing.

## JIRA

- *CNDIT-1753*: [Jira Ticket](https://cdc-nbs.atlassian.net/browse/CNDE-1753?atlOrigin=eyJpIjoiNzI3MzRkNTZkNTViNGQ0NTkyYjNhZjgzYjgwNmMzMzQiLCJwIjoiaiJ9)

## Checklist

- [x] PR focuses on a single story.
- [x] New unit tests added and ensured they pass: Procedure tested with datapoints listed in documentation. 
- [x] Service has been tested in local and it works as expected.
- [x] Documentation has been updated for this code change (if needed): [Documentation](https://cdc-nbs.atlassian.net/wiki/spaces/NM/pages/533430304/004+observation#Nested-data-JSON-column-mappings)
- [ ] Code follows the Java Coding Conventions (https://www.oracle.com/java/technologies/javase/codeconventions-programmingpractices.html).

## Types of changes

What types of changes does this PR introduces?

- [ ] Bugfix
- [x] New feature
- [ ] Breaking change

## Testing

- [ ] Does this PR has >90% code coverage?
- [ ] Is the screenshot attached for code coverage?
- [ ] Does the `gradle build` pass in your local? 
- [ ] Is the `gradle build` logs attached?

### CODE REVIEW

**Review by alog-enquizit — APPROVED** [review]
lgtm 👍

File: liquibase-service/src/main/resources/db/odse/routines/021_sp_observation_event-002.sql [hunk 1]
```diff
@@ -100,35 +100,35 @@ BEGIN
...
                                   WHERE
                                       o2.observation_uid
                                           IN (
                                           select sai.observation_uid
                                           from (
-                                                   select act1.source_act_uid as observation_uid
+                                                   select act1.source_act_uid AS observation_uid
                                                    from nbs_odse.dbo.act_relationship act1 with (nolock)
-                                                   where source_act_uid is not null and act1.target_act_uid = o.observation_uid
+                                                   where source_act_uid is not null and act1.target_act_uid = o.observation_uid and act1.record_status_cd = 'ACTIVE'
```
Comment (sveselev) [inline liquibase-service/src/main/resources/db/odse/routines/021_sp_observation_event-002.sql:116]: 👍🏻

File: liquibase-service/src/main/resources/db/odse/routines/021_sp_observation_event-002.sql [hunk 2]
```diff
@@ -319,12 +327,11 @@ BEGIN
...
                                       STRING_ESCAPE(ob.display_name, 'json') AS [ovc_display_name],
                                       ob.code AS [ovc_code],
-                                      ob.code_system_cd as [ovc_code_system_cd],
-                                      ob.code_system_desc_txt as [ovc_code_system_desc_txt],
+                                      ob.code_system_cd AS [ovc_code_system_cd],
+                                      ob.code_system_desc_txt AS [ovc_code_system_desc_txt],
                                       ob.alt_cd AS [ovc_alt_cd],
                                       ob.alt_cd_desc_txt AS [ovc_alt_cd_desc_txt],
-                                      ob.alt_cd_system_cd AS [ovc_alt_cd_system_cd],
-                                      ob.alt_cd_system_desc_txt AS [ovc_alt_cd_system_desc_txt]
```
Comment (sveselev) [inline liquibase-service/src/main/resources/db/odse/routines/021_sp_observation_event-002.sql:327]: that one should stay?

Comment (upasanapattnaik-eq) [reply liquibase-service/src/main/resources/db/odse/routines/021_sp_observation_event-002.sql]: Good catch! Updated sproc.

**Review by sveselev — APPROVED** [review]
LGTM!

**Pull request merged**